File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int getCount(string s){
        int c = 0;
        for(char x: s){
            if(
                x == 'o' ||
                x == 'O' ||
                x == 'a' ||
                x == 'A' ||
                x == 'i' ||
                x == 'I' ||
                x == 'e' ||
                x == 'E' ||
                x == 'u' ||
                x == 'U'
            ) c++;
        }
         return c;       
    }
    bool halvesAreAlike(string s) {
        string a,b;
        for(int i=0;i<s.length()/2;i++) a.push_back(s[i]);
        for(int i=s.length()/2;i<s.length();i++) b.push_back(s[i]);
        return getCount(a) == getCount(b);
    }
};

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        v = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
        n = len(s)
        a = s[:(n//2)]
        b = s[(n//2):]
        ans = 0
        for c in a:
            if c in v:
                ans += 1
        for c in b:
            if c in v:
                ans -= 1
        
        return ans == 0